File: sentimental-cash/cash.py
from cs50 import get_float

# Ask how many cents the customer is owed
cents = 0.00
# do while loop for get_float for cents
while True:
    try:
        cents = get_float("Change Owed: ")
        if cents >= 0:
            break
    except ValueError:
        continue
# variable to store cents * 100
cents_ch = round(cents * 100)


coins = 0

while cents_ch > 0:
    # if cents is greater or equal to 25 that means its a quarter
    if cents_ch >= 25:
        cents_ch -= 25
        coins += 1
    # else if cents is greater than or equal to to 10 -=10 and so on....
    elif cents_ch >= 10:
        cents_ch -= 10
        coins += 1
    elif cents_ch >= 5:
        cents_ch -= 5
        coins += 1
    else:
        cents_ch -= 1
        coins += 1
# Print total number of coins to give the customer
print(coins)

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    //Prompting the user for change owed while cents is less than 0
    int cents = 0;
    do
    {
        cents = get_int("Change Owed: ");
    }
    while (cents < 0);

    return cents;
}

int calculate_quarters(int cents)
{
    // calculate the given number of quarter
    return cents / 25;
}

int calculate_dimes(int cents)
{
    // calculate the given number of dimes
    return cents / 10;
}

int calculate_nickels(int cents)
{
    // calculate the given number of nickels
    return cents / 5;
}

int calculate_pennies(int cents)
{
    // calculate the given number of pennies
    return cents / 1;
}
